### PR TITLE
#28 Change credentials for LocalStack between environments without replicating all of my environment properties

### PR DESCRIPTION
**Pull request by carter-cundiff**

### CODE REVIEW

File: extensions/extensions-docker/aissemble-spark/src/main/resources/docker/Dockerfile [hunk 1]
```diff
@@ -37,6 +37,8 @@ RUN apt-get update -y && apt-get install --assume-yes \
   && apt-get clean \
   && ln -s /usr/bin/python${PYTHON_VERSION} /usr/bin/python
 
+RUN ln -sf /usr/bin/python3.11 /usr/bin/python3
```
Comment (ewilkins-csi) [inline extensions/extensions-docker/aissemble-spark/src/main/resources/docker/Dockerfile:40]: A: Let's put a comment explaining that spark uses `python3` to execute pyspark pipelines.

File: extensions/extensions-helm/aissemble-localstack-chart/templates/credential-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.createCredentialSecret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: remote-auth-config
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/aissemble-localstack-chart/templates/credential-secret.yaml:5]: A: I could see it being useful to configure this to a different value, if I decide to use a different name for my CI/cloud secret.

**Review by ewilkins-csi — APPROVED** [review]

File: extensions/extensions-helm/aissemble-localstack-chart/templates/credential-secret.yaml (hunk 2 above)
Comment (carter-cundiff) [reply extensions/extensions-helm/aissemble-localstack-chart/templates/credential-secret.yaml]: Added yaml config option

File: foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/SparkApplicationBaseValuesS3Migration.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    /**
+     * Function to determine whether a SparkApplication base values contains Localstack configuration that we would like to migrate.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File sparkApplicationBaseValuesFile) {
+        SparkApplication baseValuesObject = YamlUtils.createYamlObjectFromFile(sparkApplicationBaseValuesFile, SparkApplication.class, logger);
+
+        // check that the base values file contains localstack configuration
+        if (hasLocalstackConfig(baseValuesObject.getDriverEnvs()) && 
+                hasLocalstackConfig(baseValuesObject.getExecutorEnvs())) {
```
[the remark this answers is not in the document]
Comment (carter-cundiff) [reply foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/SparkApplicationBaseValuesS3Migration.java:46]: Changed logic to accommodate for either executor or driver having the hardcoded localstack credentials. Created new BDD scenarios to test.

**Pull request merged**